src/main/java/TicTacToe.java

```diff
@@ -0,0 +1,327 @@
+    }
+
+    private WinConfig isWinningConfig()
+    {
+        WinConfig w = WinConfig.WIN ;
+        // rows
+        for( int i = 0 ; i < N ; i ++ )
+        {
+            if( (board[i][0] != 0) && (board[i][0] == board[i][1]) && (board[i][0] == board[i][2] ) )
+            {
+                return w ;
+            }
+        }
+        // columns
+        for( int i = 0 ; i < N ; i ++ )
+        {
+            if( (board[0][i] != 0) && (board[0][i] == board[1][i]) && (board[0][i] == board[2][i] ) )
+            {
+                return w ;
+            }
+        }
+        // diags
+        if( (board[0][0] != 0) && (board[0][0] == board[1][1]) && (board[0][0] == board[2][2] ) )
+        {
+            return w ;
+        }
+
+        if( (board[2][0] != 0) && (board[2][0] == board[1][1]) && (board[2][0] == board[0][2] ) )
+        {
+            return w ;
+        }
+
+        // draw
+        w = WinConfig.DRAW ;
+        for( int i = 0 ; i < N ; i ++ )
+            for( int j = 0 ; j < N ; j ++ )
+                {
+                    if( board[i][j] == 0 )
+                        w = WinConfig.NONE ;
+                }
+        return w ;
+
+    }
+
+    private String getRowString(int row)
+    {
+        String s = "" ;
+        for( int i = 0 ; i < N ; i ++ )
+        {
+            switch(board[row][i]) {
+            case 0: s += " " ;
+                break ;
+            case 1: s += "O" ;
+                break ;
+            case 2: s += "X" ;
+            }
+
+            if( i != N-1 )
+                {
+                    s += " | " ;
+                }
+        }
+
+            s += String.format("%" + HSPACE + "s", "");
+
+         for( int i = 0 ; i < N ; i ++ )
+          {
+            s += row*3+i+1 ;
+
+            if( i == N-1 ) {
+                s += "\n";
+            }
+            else {
+                s += " | " ;
+            }
+          }
+        return s;
+    }
+
+    public String toString()
+    {
+        String s = "";
+        // iterate through the rows
+        for( int i = 0 ; i < N ; i ++ ) {
+            s += getRowString(i);
+        }
+        return s;
+    }
+
+    public static void main( String[] args )
+    {
+        System.out.println("Welcome to Tic-Tac-Toe.");
+        System.out.println("");
+
+        TicTacToe game = new TicTacToe();
+        String move_str ;
+        int move1 = 0 ;
+        int move2 = 0 ;
+        int player_type = 0 ;
+        WinConfig w = WinConfig.NONE ;
+
+        System.out.println("Please make your move selection by entering a number 1-9 corresponding to the movement key on the right.\n");
+        System.out.println(game.toString()) ;
+
+        while( game.isWinningConfig() == WinConfig.NONE  )
+        {
+            do {
+            move1 = game.getplayer1().getMove();
+            } while(!game.setMove(move1,game.getplayer1().getPlayerType() ));
+
+
+            if( ( w = game.isWinningConfig() ) == WinConfig.WIN ) {

```

Comment:
This is hard to read. If we were to do this best to get the config and set it, and then use the variable everywhere.